File: backend/src/main/java/com/example/starter_project_2025/system/course/entity/CourseObjective.java
package com.example.starter_project_2025.system.course.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(
        name = "course_objectives",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"course_id", "name"})
        }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CourseObjective {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    private LocalDateTime createdDate;

    @PrePersist
    public void prePersist() {
        this.createdDate = LocalDateTime.now();
    }
}

File: frontend/src/types/courseObjective.ts
export interface CourseObjective {
  id: string;
  code: string;
  name: string;
  description?: string;
  courseId: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateCourseObjectiveRequest {
  name: string;
  code: string;
  description?: string;
}

export interface UpdateCourseObjectiveRequest {
  name: string;
  code: string;
  description?: string;
}